src/components/AppNavHeader.tsx

```diff
@@ -15,25 +17,63 @@ export function AppNavHeader({
   const { styles } = useStyles(defaultStyles);
   const { ChevronLeft } = useIcons();
   const title = options.title || route.name;
+  const config = useDeveloperConfig();
+  const headerStyles = useColorMapping(
+    'backgroundColor',
+    config.AppNavHeader?.headerColors,
+    route,
+    styles.style,
+  );
+  const titleStyles = useColorMapping(
+    'color',
+    config.AppNavHeader?.onHeaderColors,
+    route,
+    styles.titleText,
+  );
 
   return (
-    <Appbar.Header style={styles.style} elevated>
+    <Appbar.Header statusBarHeight={0} style={headerStyles}>
       {back ? (
         <Appbar.Action
           icon={ChevronLeft}
+          color={styles.backActionIconAny?.color}
           onPress={navigation.goBack}
           style={styles.backAction}
         />
       ) : null}
       <Appbar.Content
-        title={<Title text={title} style={styles.titleText} />}
+        title={<Title text={title} style={titleStyles} />}
         style={styles.content}
       />
     </Appbar.Header>
   );
 }
 
-const Title = ({ text, style }: { text: string; style?: TextStyle }) => (
+const useColorMapping = (
+  property: keyof ViewStyle | keyof TextStyle,
+  routeColors: RouteColor[] | undefined,
+  route: Record<string, any>,
+  style: StyleProp<ViewStyle | TextStyle>,
+) => {
+  const theme = useTheme();
+  const title = route.params?.appTile?.title;
+  const routeColor = routeColors?.find(
+    (item) => item.route === route.name || item.route === title,
```

Comment:
This matches first on route name, then on the appTile title passed in the params.  This is because all of our app tiles have the same route name.  (React Nav 6 uses a ephemeral generated id to keep track of routes internally.)

Previously we matched on app tile id, but this is problematic as they will change when a new app tile is published.